packages/eas-cli/src/commands/branch/publish.ts

```diff
@@ -122,7 +123,16 @@ export default class BranchPublish extends Command {
 
     const { exp } = getConfig(projectDir, { skipSDKVersionRequirement: true });
     const accountName = await getProjectAccountNameAsync(exp);
-    const { slug, runtimeVersion } = exp;
+    let { slug, runtimeVersion, sdkVersion } = exp;
+
+    // When a SDK version is supplied instead of a runtime version and we're in the managed workflow
+    // construct the runtimeVersion with special meaning indicating that the runtime is an
+    // Expo SDK preset runtime that can be launched in Expo Go.
+    const isManagedProject = getDefaultTarget(projectDir) === 'managed';
+    if (!runtimeVersion && sdkVersion && isManagedProject) {
+      runtimeVersion = getRuntimeVersionForSDKVersion(sdkVersion);
```

Comment:
Is it worth keeping the user informed about what is happening with a:

`Log.withTick('Generating runtime version from sdk version')'`